File: problems/136_single-number/main.cpp
#include <iostream>
#include <vector>

using namespace std;

class Solution
{
public:
    int singleNumber(vector<int> &nums)
    {
        int single = 0;
        for (auto i : nums)
            single = single ^ i;
        return single;
    }
};

int main()
{
    Solution solution;

    vector<vector<int>> test_cases = {
        {2, 2, 1},
        {4, 1, 2, 1, 2},
        {1}};
    vector<int> expected = {1, 4, 1};

    for (size_t i = 0; i < test_cases.size(); ++i)
    {
        vector<int> nums = test_cases[i];
        int result = solution.singleNumber(nums);
        cout << "Sample " << (i + 1) << ": result=" << result
             << ", expected=" << expected[i] << '\n';
    }

    return 0;
}

File: problems/136_single-number/main.rs
struct Solution;

impl Solution {
    pub fn single_number(nums: Vec<i32>) -> i32 {
        let mut single = 0;
        for &i in &nums {
            single ^= i;
        }
        single
    }
}

fn main() {
    let test_cases = vec![vec![2, 2, 1], vec![4, 1, 2, 1, 2], vec![1]];
    let expected = [1, 4, 1];

    for (idx, nums) in test_cases.into_iter().enumerate() {
        let result = Solution::single_number(nums);
        println!(
            "Sample {}: result={}, expected={}",
            idx + 1,
            result,
            expected[idx]
        );
    }
}
